modular_skyrat/modules/goofsec/code/projectiles.dm

```diff
@@ -201,3 +201,279 @@
 	damage = 60
 */
 
+/// Bullet security pistol.
+/obj/item/gun/ballistic/automatic/pistol/security
+	name = "security pistol"
+	desc = "A pistol issued to station security personnel. It's a small, compact pistol, but it packs a punch."
+	mag_type = /obj/item/ammo_box/magazine/security_pistol
+	icon = 'modular_skyrat/modules/sec_haul/icons/guns/ladon.dmi'
+	righthand_file = 'modular_skyrat/modules/sec_haul/icons/guns/inhands/righthand40x32.dmi'
+	lefthand_file = 'modular_skyrat/modules/sec_haul/icons/guns/inhands/lefthand40x32.dmi'
+	icon_state = "ladon"
+	inhand_icon_state = "ladon"
+	can_suppress = FALSE
+	fire_sound = 'modular_skyrat/modules/sec_haul/sound/pistol_fire.ogg'
+	rack_sound = 'sound/weapons/gun/pistol/rack.ogg'
+	lock_back_sound = 'sound/weapons/gun/pistol/slide_lock.ogg'
+	bolt_drop_sound = 'sound/weapons/gun/pistol/slide_drop.ogg'
+
+/obj/item/ammo_box/magazine/security_pistol
+	name = "influence security pistol magazine"
+	desc = "A magazine for the security pistol. Loaded with rubberized influence rounds."
+	icon = 'modular_skyrat/modules/goofsec/icons/gun_sprites.dmi'
+	icon_state = "security_pistol_influence"
+	base_icon_state = "security_pistol_influence"
+	ammo_type = /obj/item/ammo_casing/security_pistol/influence
+	caliber = CALIBER_SECURITYPISTOL
+	max_ammo = 20
+
+/obj/item/ammo_box/magazine/security_pistol/update_icon_state()
+	. = ..()
+	icon_state = "[base_icon_state][ammo_count() ? "" : "-empty"]"
+
+/// Ammo for the Security Pistol.
+/obj/item/ammo_casing/security_pistol
```

Comment:
redundant basetype, make the basetype do something.

Reply:
the basetype isn't redundant, it's being used to reduce copypaste of variables shared across all of the bullets and casings